### ISSUE TITLE
Bad default for jobs flag ignores user config

### ISSUE THREAD

**Issue by ScottBailey** [AntelopeIO/antler-proj#28]
In files:
- include/antler/project/net_utils.hpp
- include/antler/project/location.hpp
- include/antler/project/populator.hpp

The default argument value of `1` for the various `jobs` parameters feels wrong on any modern multi-core machine. Additionally, as it's currently implemented the user's `submodule.fetchJobs` configuration is ignored. This doesn't seem very friendly or optimal.

Please consider one of the following options:
- default to `0` and allow git to select its own job count, still ignores user settings.
- default to `std::numeric_limits<T>::max()` and then ONLY call git with a value when actually set.  
- remove entirely and allow git's default behavior, including honoring the user's config, to take over.
- remove entirely and allows call git with `--jobs=0` UNLESS the user has configured `submodule.fetchJobs`

**Comment by dimas1185** [comment]
I think we should leave that to git. so `--jobs=0` looks like decent solution

**Comment by ScottBailey** [comment]
Based on this [commit](https://github.com/git/git/commit/d180cc297922569e530da843bee4df3964167dea), I do not think sending `--jobs=0` to `git clone` is a good idea.

Our currently targeted OSes do not included the updated version, 2.40.

**Comment by ScottBailey** [comment]
As implemented in main, the `--jobs` flag is superfluous - it does nothing.

https://github.com/AntelopeIO/antler-proj/blob/main/include/antler/project/net_utils.hpp#L190-L202

```cpp

      /// @brief clone a repo from git
      /// @param url
      /// @param branch
      static bool clone(const std::string& url, const std::string& branch, uint32_t jobs, const system::fs::path& dest) {
         int32_t ret = system::execute(std::string(executable), { "clone", "-j", std::to_string(jobs), url, "--depth", "1", "--branch", branch, dest.string() });
         system::debug_log("clone for {0} returned {1}\n", url,ret);
         if (ret != 0)
            return false;
         ret = system::execute(std::string(executable), { "-C", dest.string(), "submodule", "sync" });
         system::debug_log("sync for {0} returned {1}\n", url, ret);
         if (ret != 0)
            return false;
         ret = system::execute(std::string(executable), { "-C", dest.string(), "submodule", "update", "--init", "--recursive" });
         system::debug_log("submodule update for {0} returned {1}\n", url, ret);
         return ret == 0;
      }
```

Looking at git's documentation and source code, it seems that parallel downloading only happens for submodules. 

In the above code, you can see jobs is sent to `clone` but lacks the `--recurse-submodules` flag. And the `submodule` command does not receive the `--jobs` flag.

I think it would be all together better to simply remove the jobs flag and point the user to git's `submodule.fetchJobs` [documentation](https://git-scm.com/docs/git-config#Documentation/git-config.txt-submodulefetchJobs) if we feel the need for parallelization of submodule fetches.

Generally speaking, simpler code is better. We have an opportunity to make this simpler without losing functionality.
